rust/lance/src/io/exec/knn.rs

```diff
@@ -425,7 +425,12 @@ impl DisplayAs for KNNIndexExec {
     fn fmt_as(&self, t: DisplayFormatType, f: &mut std::fmt::Formatter) -> std::fmt::Result {
         match t {
             DisplayFormatType::Default | DisplayFormatType::Verbose => {
-                write!(f, "KNNIndex: name={}, k={}", self.index.uuid, self.query.k)
+                write!(
+                    f,
+                    "KNNIndex: name={}, k={}",
+                    self.index.uuid,
+                    self.query.k * self.query.refine_factor.unwrap_or(1) as usize
```

Comment:
I get what you are after but this might be slightly confusing since `k` is used elsewhere.  Maybe change `k={}` to `num_rows={}`?

Reply:
I personally find it clear in-context of a plan:

https://github.com/lancedb/lance/blob/ee1fb9aa8e4bd4989678588889ecc916211a6cd9/rust/lance/src/dataset/scanner.rs#L3217-L3221

We see that originally k=40 and then later we take k=10 with exact, which is a good representation of what `refine_factor=4` means. TBH I was tempted to refactor `KNNIndex` to not have a `refine_factor` parameter and just have a `k` parameter that was set to the query `k * refine_factor`. `KNNIndex` shouldn't need to know about `refine_factor`.

Reply:
Ok, I agree with that reasoning.